packages/compiler-cli/ngcc/src/host/umd_host.ts

```diff
@@ -547,51 +551,216 @@ function getUmdWrapper(statement: ts.Statement):
+  // whose one side is a `typeof exports` or `typeof module` condition.
+  const cjsConditionalCall = calls.find(
+      call => call.condition?.operatorToken.kind === ts.SyntaxKind.AmpersandAmpersandToken &&
+          oneOfBinaryConditions(call.condition, exp => isTypeOf(exp, 'exports', 'module')) &&
+          ts.isIdentifier(call.factoryCall.expression) &&
+          call.factoryCall.expression.text === 'factory');
+
+  return cjsConditionalCall?.factoryCall ?? null;
+}
+
+/**
+ * Get the factory call for setting up the global dependencies in the UMD wrapper.
+ */
+function getGlobalFactoryCall(calls: UmdConditionalFactoryCall[]): ts.CallExpression|null {
+  // The factory call for global dependencies is the one that is the final else-case (i.e. the one
+  // that has `condition: null`).
+  const globalConditionalCall = calls.find(call => call.condition === null);
+
+  return globalConditionalCall?.factoryCall ?? null;
+}
+
+function oneOfBinaryConditions(
+    node: ts.BinaryExpression, test: (expression: ts.Expression) => boolean) {
+  return test(node.left) || test(node.right);
+}
+
+function isTypeOf(node: ts.Expression, ...types: string[]): boolean {
+  return ts.isBinaryExpression(node) && ts.isTypeOfExpression(node.left) &&
+      ts.isIdentifier(node.left.expression) && types.includes(node.left.expression.text);
+}
 
 export function getImportsOfUmdModule(umdModule: UmdModule):
     {parameter: ts.ParameterDeclaration, path: string}[] {
   const imports: {parameter: ts.ParameterDeclaration, path: string}[] = [];
+  const cjsFactoryCall = umdModule.factoryCalls.commonJs;
+
   for (let i = 1; i < umdModule.factoryFn.parameters.length; i++) {
     imports.push({
       parameter: umdModule.factoryFn.parameters[i],
-      path: getRequiredModulePath(umdModule.wrapperFn, i)
+      path: getRequiredModulePath(cjsFactoryCall, i),
     });
   }
+
   return imports;
 }
 
 interface UmdModule {
   wrapperFn: ts.FunctionExpression;
   factoryFn: ts.FunctionExpression;
+  factoryCalls: {
+    commonJs: ts.CallExpression; amdDefine: ts.CallExpression | null;
+    global: ts.CallExpression | null;
+  };
+}
+
+/**
+ * Represents a factory call found inside the UMD wrapper function.
+ *
+ * Each factory call corresponds to a format (such as AMD, CommonJS, etc.) and is guarded by a
+ * condition (except for the last factory call, which is reached when all other conditions fail).
+ */
+interface UmdConditionalFactoryCall {
+  condition: ts.BinaryExpression|null;
+  factoryCall: ts.CallExpression;
 }
 
-function getRequiredModulePath(wrapperFn: ts.FunctionExpression, paramIndex: number): string {
-  const statement = wrapperFn.body.statements[0];
-  if (!ts.isExpressionStatement(statement)) {
+function getRequiredModulePath(cjsFactoryCall: ts.CallExpression, paramIndex: number): string {
+  const requireCall = cjsFactoryCall.arguments[paramIndex];
+
+  if (!isRequireCall(requireCall)) {
```

Comment:
WDYT about safe-guarding against the argument for index `paramIndex` missing:

```suggestion
  if (requireCall !== undefined && !isRequireCall(requireCall)) {
```

Reply:
Good idea :+1: